cmake/ExperimentalEmscripten.cmake

```diff
@@ -0,0 +1,112 @@
+# 8 threads deadlocked consistently, 16 threads passed consistently. 
+# 32 was chosen to give enough headroom for tests written in the future. 
+#
+# `-sPROXY_TO_PTHREAD`
+# This flag wraps our executable's main function in a pthread. 
+# Without this, we exhaust the thread pool very quickly. The error looks like this:
+#
+#   Tried to spawn a new thread, but the thread pool is exhausted.
+#   This might result in a deadlock unless some threads eventually exit or the code explicitly breaks out to the event loop.
+# 
+# You can read more about the setting here: https://emscripten.org/docs/porting/pthreads.html#blocking-on-the-main-browser-thread
+#
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -fwasm-exceptions -sSUPPORT_LONGJMP=wasm")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pthread -fwasm-exceptions -sEXIT_RUNTIME=1 -sINITIAL_MEMORY=300MB -sPTHREAD_POOL_SIZE=32 -sPROXY_TO_PTHREAD")
+set(STLAB_WASM_NODE_JS_MIN_VERSION "16.16.0")
+
+#
+# Note: https://www.npmjs.com/package/wasm-check is a useful utility to find 
+# which --experimental-wasm-xxx flags are supported by node.
+#
+set(STLAB_REQUIRED_WASM_EXPERIMENTAL_FLAGS "--experimental-wasm-threads;--experimental-wasm-eh")
+
+#
+# Check if NODE_JS_EXECUTABLE (found by find_program() in Emscripten.cmake) is recent enough for STLab.
+# Set CMAKE_CROSSCOMPILING_EMULATOR to a sufficiently recent node + required experimental flags.
+#
+if (NODE_JS_EXECUTABLE)
+    message(STATUS "stlab:wasm: Ensuring ${NODE_JS_EXECUTABLE} is at least ${STLAB_WASM_NODE_JS_MIN_VERSION}...")
+    execute_process( COMMAND ${NODE_JS_EXECUTABLE} --version OUTPUT_VARIABLE NODE_JS_EXECUTABLE_V_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+    STRING(REPLACE "v" "" NODE_JS_EXECUTABLE_VERSION ${NODE_JS_EXECUTABLE_V_VERSION})
+
+    if (NODE_JS_EXECUTABLE_VERSION VERSION_GREATER_EQUAL ${STLAB_WASM_NODE_JS_MIN_VERSION})
+        message(STATUS "stlab:wasm: Installed node satisfies requirements: ${NODE_JS_EXECUTABLE_VERSION}")
+        set(CMAKE_CROSSCOMPILING_EMULATOR "${NODE_JS_EXECUTABLE};${STLAB_REQUIRED_WASM_EXPERIMENTAL_FLAGS}")
+    else()
+        message(FATAL_ERROR "stlab:wasm: Unsupported node: ${NODE_JS_EXECUTABLE_VERSION}. Please install ${STLAB_WASM_NODE_JS_MIN_VERSION} or newer.")
+    endif()
+else()
+    message(FATAL_ERROR "stlab:wasm: Unable to find node. Please install ${STLAB_WASM_NODE_JS_MIN_VERSION} or newer.")
+endif()
+
+#
+# Emscripten supports dynamic linking, but doing so introduces some complexity:
+# https://emscripten.org/docs/compiling/Dynamic-Linking.html
+# We presently have no need to dynamically link WASM modules, so we instruct
+# boost to link statically.
+# 
+# It would be nice to set a more generic variable here - like (BUILD_SHARED_LIBS OFF)
+# which any library could check to set their own corresponding variable.
+# But, Boost doesn't consider BUILD_SHARED_LIBS when setting this default.
+#
+set(Boost_USE_STATIC_LIBS ON)
```

Comment:
For good measure, I think it would be good to also add,

```cmake
set( BUILD_SHARED_LIBS FALSE )
```

, to this toolchain file. That's an appropriate default.